gear/gear/clients.py

```diff
@@ -23,21 +23,6 @@ def get_identity_client(credentials_file: Optional[str] = None):
     return aiogoogle.GoogleIAmClient(project, credentials_file=credentials_file)
 
 
-def get_compute_client(credentials_file: Optional[str] = None):
```

Comment:
What is your rationale for refactoring this code? I thought it would be good to have all of the cloud abstractions for getting a compute client in the same place as the storage client. Otherwise, I thought it would be prone to error to have this same code called in multiple places.

Reply:
> I thought it would be good to have all of the cloud abstractions for getting a compute client in the same place as the storage client.

I think I personally don't see the benefit here. I'm not sure what you mean by "this same code", or why the storage and compute client getters need to be colocated. If there's some similarity between the two, I'd rather codify it in the types of the code rather than the position of the code. `get_compute_client` is only called in the worker process, so I don't see a reason at the moment for this code to be at a higher level of visibility than that.

However, my main motivations for moving this are the following two reasons:

1. I personally find this pattern of repeatedly checking `get_global_config()['cloud']` very noisy. It increases the surface area that this environment variable reaches and in the case of the worker it is mixing two design choices:
-  Having an object-oriented WorkerAPI class that knows cloud-specific implementation details
- A more function-based approach like this one that switches on global state to produce different implementations

I'd rather have only one of these approaches, and I like the former in the case of the worker because `worker.py` is already very object-oriented and because we only have to check the cloud once. When I trace the flow of the program from the worker main method through to some small detail, it is very noisy to me to be switching between cloud-specific and cloud-agnostic methods in the same callstack.

2. I wanted to share the cloud credentials object between the FS and the compute client. I think it would be awkward to introduce a `CloudSession` input to this method because it would involve an upcast followed by a downcast, which worsens my issue with multiple spots on the same codepath trying to figure out what cloud we're in. So at that point, I didn't see much value in having the function and stuck the appropriate parts of this code closer to where the corresponding inputs are.

Reply:
ok if this makes more sense to you. I think if we have multiple applications (not just batch) in the future that need a compute client, then we can revisit what the right abstraction is.